sycl/doc/design/HostPipes.md

```diff
@@ -0,0 +1,188 @@
+# Implementation design for "Host Pipes"
+
+This document describes the implementation design for the host pipes section
+of the DPC++ extension [SYCL_INTEL_data_flow_pipes][1]. Pipes are a FIFO construct 
+that provide links between elements of a design that are accessed through read 
+and write application programming interfaces (APIs), without the notion of a
+memory address/pointer to elements within the FIFO. A host pipe is a pipe that 
+links a device kernel with a host program.
+
+[1]: https://github.com/intel-sandbox/ip-authoring-specs/blob/main/Pipe/Spec/data_flow_pipes.asciidoc
+
+## Requirements
+
+The extension specification document referenced above contains the full set of
+requirements for this feature, but a requirement that is particularly
+relevant to the design, and similar in nature to one raised in the [device_global][2]
+design is called out here.
+
+This issue relates to the mechanism for integrating host and device code.
+Like device global variables, host pipes are referenced in both
+host and device code, so they require some mechanism to correlate the pipe
+instance in device code with the pipe instance in host code. We will use
+a similar mechanism as the device global implementation that creates a map
+database in the integration headers and footers.
+
+[2]: <DeviceGlobal.md>
+
+## Design
+
+### Changes to DPC++ headers
+
+#### Attributes attached to the class
+
+The `pipe` class uses a new C++ attribute `[[__sycl_detail__::host_pipe]]` on the 
+`pipe::__pipeType` type to identify the `static const __pipeType` member `__pipe`
+as a host pipe. Similar to `[[__sycl_detail__::device_global]]`, this will inform
+the front end to generate a `sycl-unique-id` for each `__pipe`. The `pipe` class
+also introduces the global variable attribute `sycl-host-pipe` attribute to inform the sycl-post-link tool
+to generate the SPIR-V decoration `HostAccessINTEL` for each `__pipe` using the
+`sycl-unique-id` generated. 
+
+As these attributes are only needed for the device compiler, the `#ifdef __SYCL_DEVICE_ONLY__` 
+allows the customer to ue another host compiler, even if it does not recognize these attributes.
+Also note that these attributes are all in the `__sycl_detail__` namespace, so
+they are considered implementation details of DPC++.  We do not intend to
+support them as general attributes that customer code can use.
+
+```
+template <typename name,
+          typename dataT,
+          typename propertiesT = properties<>>
+class pipe
+{ 
+  struct
+#ifdef __SYCL_DEVICE_ONLY__
+  [[__sycl_detail__::add_ir_global_variable_attributes(
+    "sycl-host-pipe",
+    nullptr
+    )]]
```

Comment:
The attribute has been renamed to [`add_ir_attributes_global_variable`](https://github.com/intel/llvm/blob/sycl/sycl/doc/design/CompileTimeProperties.md#properties-on-a-global-variable-type), which I now realize also need to be changed in the [`device_global` design document](https://github.com/intel/llvm/blob/sycl/sycl/doc/design/DeviceGlobal.md). I see that the document uses the new name further down, so I imagine this was just an oversight.

I assume there may be compile-time properties that need to be communicated to the backend. If that is the case, the use of `__sycl_detail__::add_ir_attributes_global_variable` attribute should expand the properties to their `meta_name` and `meta_value`, like in the [`device_global` design document](https://github.com/intel/llvm/blob/sycl/sycl/doc/design/DeviceGlobal.md).
Note that I plan on changing the way these are extracted slightly (see https://github.com/intel/llvm/pull/5959) but if you go with the same approach as in the current design document I will make the amendment if this is merged first.